NLL: Wrong variable name in E0502 (double borrow) error message

On nightly, the following code...

```rust
#![feature(nll)]

struct Bar;

impl Bar {
    fn bar(&mut self, _: impl Fn()) { }
}

struct Foo { thing: Bar, number: usize }

impl Foo {
    fn foo(&mut self) {
        self.thing.bar(|| {
            &self.number;
        });
    }
}
```

...yields this error message:

```none
error[E0502]: cannot borrow `self.thing` as mutable because it is also borrowed as immutable
  --> src/lib.rs:13:9
   |
13 |            self.thing.bar(|| {
   |            ^              -- immutable borrow occurs here
   |   _________|
   |  |_________|
   | ||
14 | ||             &self.number;
   | ||              ---- previous borrow occurs due to use of `self.thing` in closure
15 | ||         });
   | ||          ^
   | ||__________|
   | |___________mutable borrow occurs here
   |             borrow later used here
```

Since the closure actually borrows `self.number`, not `self.thing`, I feel like the error should say `self` instead of `self.thing`. It would also be nice if it weren't an error in the first place, but that's a separate issue (let me know if I should report it separately).

---

## Meta

`rustc --version --verbose`:
```none
rustc 1.28.0-nightly (1ffb32147 2018-05-31)
binary: rustc
commit-hash: 1ffb321477d85fb3c95a0be2a0cd89c04dff0d8d
commit-date: 2018-05-31
host: x86_64-unknown-linux-gnu
release: 1.28.0-nightly
LLVM version: 6.0
```

--- a/src/librustc_mir/borrow_check/error_reporting.rs
+++ b/src/librustc_mir/borrow_check/error_reporting.rs
@@ -364,6 +364,9 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
         };
 
         if let Some((_, var_span)) = old_closure_span {
+            let place = &issued_borrow.borrowed_place;
+            let desc_place = self.describe_place(place).unwrap_or("_".to_owned());
+
             err.span_label(
                 var_span,
                 format!(

--- a/src/test/ui/nll/issue-51268.rs
+++ b/src/test/ui/nll/issue-51268.rs
@@ -1,0 +1,35 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// ignore-tidy-linelength
+
+#![feature(nll)]
+
+struct Bar;
+
+impl Bar {
+    fn bar(&mut self, _: impl Fn()) {}
+}
+
+struct Foo {
+    thing: Bar,
+    number: usize,
+}
+
+impl Foo {
+    fn foo(&mut self) {
+        self.thing.bar(|| {
+        //~^ ERROR cannot borrow `self.thing` as mutable because it is also borrowed as immutable [E0502]
+            &self.number;
+        });
+    }
+}
+
+fn main() {}

--- a/src/test/ui/nll/issue-51268.stderr
+++ b/src/test/ui/nll/issue-51268.stderr
@@ -1,0 +1,20 @@
+error[E0502]: cannot borrow `self.thing` as mutable because it is also borrowed as immutable
+  --> $DIR/issue-51268.rs:28:9
+   |
+LL |            self.thing.bar(|| {
+   |            ^              -- immutable borrow occurs here
+   |   _________|
+   |  |_________|
+   | ||
+LL | ||         //~^ ERROR cannot borrow `self.thing` as mutable because it is also borrowed as immutable [E0502]
+LL | ||             &self.number;
+   | ||              ---- previous borrow occurs due to use of `self` in closure
+LL | ||         });
+   | ||          ^
+   | ||__________|
+   | |___________mutable borrow occurs here
+   |             borrow later used here
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0502`.

